library/src/blas_ex/rocblas_gemm_ex.hpp

```diff
@@ -743,63 +667,32 @@ rocblas_status gemm_ex_handle_transpose(rocblas_handle    handle,
     }
     else
     {
-        device_strided_batched_matrix_copy(
-            c, ldc, stride_c, d, ldd, stride_d, m, n, batch_count, sizeof(To));
+        device_strided_batched_matrix_copy(c, ldc, stride_c, d, ldd, stride_d, m, n, batch_count);
         c_in     = d;
         ldi      = ldd;
         stride_i = stride_d;
```

Comment:
Potential data loss moving to smaller type?    Looks like tensile doesn't support the 64bit int for stride either?

Reply:
> Potential data loss moving to smaller type? Looks like tensile doesn't support the 64bit int for stride either?

We may need to file a Tensile bug then, if it does not support 64-bit stride.

There are many places in rocBLAS where `int` types and sizes are used carelessly, such as using `int` instead of `rocblas_int` for a loop variable whose limits are `rocblas_int`, or not converting to `size_t` before doing a size calculation with multiplication. We need to go through and sweep rocBLAS looking for size calculations which may truncate or overflow.

The change referred to here was only `clang-format` changes after the removal of a `sizeof(To)` argument in a call, and instead making `To` a template parameter automatically deduced from the `c` and `d` pointers.

Formerly:
```
void func(..., size_t elemsize)
{
    size_t size = elemsize;
}

func(..., c, ..., sizeof(To)):
```
New:
```
template <typename To>
void func(..., To *c, ...)
{
   size_t  size =  sizeof(To);
}

func(..., c, ...);
```
So the change does not introduce any new truncation from 64-bit to 32-bit. That problem already existed and will remain until Tensile supports 64-bits, which is beyond the scope of this PR.

Reply:
Tensile uses size_t for all strides right now.  Not sure where you got that it doesn't support 64-bit strides?

We may be moving to int64_t at some point to allow specifying -1 as an "auto" stride.

Reply:
> 
> 
> Tensile uses size_t for all strides right now. Not sure where you got that it doesn't support 64-bit strides?
> 
> We may be moving to int64_t at some point to allow specifying -1 as an "auto" stride.

That's what I thought; right now `rocblas_stride` is defined as `int64_t` but it could just as well be `size_t` -- both are the same size, and conversion between them is a no-op in 2's complement.

Reply:
Sorry my comment on stride related to the "tensile" calls in code context right above the lines of code I highlighted (not to the tensile library in general).  E.g. code #define pasted below which I think has already been changed.  32bit integers were being used for strides.  We were discussing negative strides to allow reverse ordering in rocblas so I think this may still be a consideration we need to iron out with tensile, otherwise I had no concern to raise with tensile itself.      

#define TENSILE_IN_ARGS(Ti, To, Tc)                                                         \
    To *dataD, const To *dataC, const Ti *dataA, const Ti *dataB, Tc alpha, Tc beta,             \
    To *dataD, const To *dataC, const Ti *dataA, const Ti *dataB, Tc alpha, Tc beta,        \
        unsigned int strideD1J, unsigned int strideD2K, unsigned int strideC1J,                  \
        unsigned strideD1J, unsigned strideD2K, unsigned strideC1J, unsigned strideC2K,     \
        unsigned int strideC2K, unsigned int strideA1L, unsigned int strideA2K,                  \
        unsigned strideA1L, unsigned strideA2K, unsigned strideB1J, unsigned strideB2K,     \
        unsigned int strideB1J, unsigned int strideB2K, unsigned int sizeI, unsigned int sizeJ,  \
        unsigned sizeI, unsigned sizeJ, unsigned sizeK, unsigned sizeL, hipStream_t stream, \
        unsigned int sizeK, unsigned int sizeL, hipStream_t stream, unsigned int numInputEvents, \
        unsigned numInputEvents, void *dummy1, void *dummy2

Reply:
> We were discussing negative strides to allow reverse ordering in rocblas so I think this may still be a consideration we need to iron out with tensile, otherwise I had no concern to raise with tensile itself.

We already decided that negative strides should be handled just like any other in the kernel code, and that they should not be handled like `incx` and `incxy` are handled in `AXPY`, `SWAP`, etc.

Negative strides in the test code simply involves handling a mismatch between the stated problem and the pointers returned by allocated memory: The stated problem is one with a starting address and negative strides, while the pointers returned for allocated memory always start at the lower address. Hence we need to adjust the pointers to the end of allocated memory, if the problem has negative strides. But the rocBLAS and Tensile kernels do not need to do anything special -- they simply take a starting address and a stride, which can be positive or negative.

The macros you show above are from the **old** Tensile client code, which was probably never fully ported to use the new `rocblas_stride` type. The **new** Tensile client code always represents strides using `rocblas_stride`, which is defined as `int64_t` right now.